packages/utils/src/PromiseUtils.ts

```diff
@@ -1,16 +1,20 @@
 import CanceledPromiseError from './CanceledPromiseError';
 import TimeoutError from './TimeoutError';
 
+interface CancelablePromise<T> extends Promise<T> {
+  cancel: () => void;
+}
+
 class PromiseUtils {
-  static isCancelablePromise(promise) {
+  static isCancelablePromise(promise: CancelablePromise<unknown>): boolean {
     return (
       this.isPromise(promise) &&
       promise.cancel != null &&
       typeof promise.cancel === 'function'
     );
   }
 
-  static isPromise(promise) {
+  static isPromise(promise: Promise<unknown>): boolean {
```

Comment:
Same thing about using a type predicate.

You might have to cast the values for these such as `(promise as Promise<unknown>).then != null`

```suggestion
  static isPromise(promise: unknown): promise is Promise<unknown> {
```